Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1374,16 +1374,17 @@ bool GCMemcard::Format(u8* card_data, const CardFlashId& flash_id, u16 size_mbit
 {
   if (!card_data)
     return false;
-  memset(card_data, 0xFF, BLOCK_SIZE * 3);
-  memset(card_data + BLOCK_SIZE * 3, 0, BLOCK_SIZE * 2);
 
-  *((Header*)card_data) =
-      Header(flash_id, size_mbits, shift_jis, rtc_bias, sram_language, format_time);
+  Header header(flash_id, size_mbits, shift_jis, rtc_bias, sram_language, format_time);
+  Directory dir;
+  BlockAlloc bat(size_mbits);
+
+  std::memcpy(&card_data[BLOCK_SIZE * 0], &header, BLOCK_SIZE);
```

Comment:
Shouldn't this be `sizeof(Header)` instead of `BLOCK_SIZE` ?

Reply:
All blocks on memory cards are the same size so it doesn't really matter. I think it's clearer this way due to the multiplication to the left but ymmv.

https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/HW/GCMemcard/GCMemcard.h#L250
https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/HW/GCMemcard/GCMemcard.h#L364
https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/HW/GCMemcard/GCMemcard.h#L399

Reply:
Ah, I was wondering about this but didn't look close enough.  Thank you!